### ISSUE TITLE
[binami/minio] ERROR Unable to initialize console server: Specified port is already in use

### ISSUE THREAD

**Issue by slax81** [bitnami/charts#7757]
**Which chart**:
bitnami/minio

**Describe the bug**
On installing helm chart I get ERROR Unable to initialize console server: Specified port is already in use
      > Please ensure no other program uses the same address/port (edited) 
One time it worked on version from 06.10.21, but when I redeployed with the changed pvc size, the same error was there. I also trying to use the latest version (now is oct 10.) and I still get the same error. I don't think that the ports 9000 and 9001 are in use by anything.

**To Reproduce**

Install the helm chart with edited values.

**Expected behavior**

A woking deployment.

**Version of Helm and Kubernetes**:

latest and 06.10.21

- Output of `helm version`:

NAME: minio
LAST DEPLOYED: Sat Oct  9 15:04:02 2021
NAMESPACE: minio
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
** Please be patient while the chart is being deployed **

MinIO&reg; can be accessed via port  on the following DNS name from within your cluster:

   minio.minio.svc.cluster.local

To get your credentials run:

   export ACCESS_KEY=$(kubectl get secret --namespace minio minio -o jsonpath="{.data.access-key}" | base64 --decode)
   export SECRET_KEY=$(kubectl get secret --namespace minio minio -o jsonpath="{.data.secret-key}" | base64 --decode)

To connect to your MinIO&reg; server using a client:

- Run a MinIO&reg; Client pod and append the desired command (e.g. 'admin info'):

   kubectl run --namespace minio minio-client \
     --rm --tty -i --restart='Never' \
     --env MINIO_SERVER_ACCESS_KEY=$ACCESS_KEY \
     --env MINIO_SERVER_SECRET_KEY=$SECRET_KEY \
     --env MINIO_SERVER_HOST=minio \
     --image docker.io/bitnami/minio-client:latest -- admin info minio

To access the MinIO&reg; web UI:

- Get the MinIO&reg; URL:

   You should be able to access your new MinIO&reg; web UI through

   http://minio-console.getbamboo.com/minio/
WARNING: Rolling tag detected (bitnami/minio:latest), please note that it is strongly recommended to avoid using rolling tags in a production environment.
+info https://docs.bitnami.com/containers/how-to/understand-rolling-tags-containers/
WARNING: Rolling tag detected (bitnami/minio-client:latest), please note that it is strongly recommended to avoid using rolling tags in a production environment.
+info https://docs.bitnami.com/containers/how-to/understand-rolling-tags-containers/

- Output of `kubectl version`:

$ kubectl logs minio-5658c4c96d-455k4 -n minio
 22:07:58.65 
 22:07:58.65 Welcome to the Bitnami minio container
 22:07:58.65 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-minio
 22:07:58.65 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-minio/issues
 22:07:58.65 
 22:07:58.66 INFO  ==> ** Starting MinIO setup **
 22:07:58.67 DEBUG ==> Validating settings in MINIO_* env vars..
 22:07:58.68 INFO  ==> Starting MinIO in background...
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD

 You are running an older version of MinIO released 2 days ago 
 Update: Run `mc admin update` 

API: http://10.20.210.45:9000  http://127.0.0.1:9000 

Console: http://10.20.210.45:9001 http://127.0.0.1:9001 

Documentation: https://docs.min.io
 22:08:08.69 INFO  ==> Adding local Minio host to 'mc' configuration...
 22:08:08.78 INFO  ==> Stopping MinIO...
Stopping on service signal

 22:08:08.87 INFO  ==> ** MinIO setup finished! **
 22:08:08.89 INFO  ==> ** Starting MinIO **
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD

 You are running an older version of MinIO released 2 days ago 
 Update: Run `mc admin update` 

API: http://10.20.210.45:9000  http://127.0.0.1:9000 

Console: http://10.20.210.45:9001 http://127.0.0.1:9001 

Documentation: https://docs.min.io
ERROR Unable to initialize console server: Specified port is already in use
      > Please ensure no other program uses the same address/port

**Additional context**

**Comment by slax81** [comment]
After several hours and 167 pod crashes, the pod started and now API and console works as expected. Nothing was changed on the cluster to

**Comment by migruiz4** [comment]
@slax81 The zombie process appears wether MinIO is able to start or not. I suspect that, in some cases first MinIO initialization would take a bit longer to free the TCP sockets 9000 and 9001, causing the second initialization to trigger the `Unable to initialize console server: Specified port is already in use`.

@datadidit Thank you for sharing your workaround, I would like to add that by setting `MINIO_SKIP_CLIENT = yes`, you will be skipping the configuration of the MinIO client and the creation of the default buckets (`MINIO_DEFAULT_BUCKETS`):

https://github.com/bitnami/bitnami-docker-minio/blob/b92073610f58ede066c8433b7a5151beaf2bc19c/2021/debian-10/rootfs/opt/bitnami/scripts/minio/setup.sh#L27-L52

https://github.com/bitnami/bitnami-docker-minio/blob/ac2778a9796c75fa26a7df17f02e94809279cbb3/2021/debian-10/rootfs/opt/bitnami/scripts/libminioclient.sh#L101-L104

https://github.com/bitnami/bitnami-docker-minio/blob/7348815cbb325ee722d91296437418522f9b9a7e/2021/debian-10/rootfs/opt/bitnami/scripts/libminio.sh#L270-L291

**Comment by migruiz4** [comment]
That is great! Thank you for letting me know that the fix is working.

I will then proceed to close this issue. If you find any further issues please let us know.

**Comment by ibrt** [comment]
I'm experiencing this issue again on a Apple Silicon M1 MacBook Pro... Same exact symptoms, doesn't happen with `MINIO_SKIP_CLIENT=yes`. When the setup script exits and Minio is restarted it fails to bind the port. Either the setup script is not properly waiting for clean exit, or there's some delay between that and when the port is fully released by the network stack.

This workaround consistently solves the problem:

```Dockerfile
FROM bitnami/minio:2023.6.23
RUN sed -i 's/setup\.sh/setup\.sh; sleep 10/g' /opt/bitnami/scripts/minio/entrypoint.sh
```

**Comment by stockmind** [comment]
Be careful if you have a registered a notification service like with `MINIO_NOTIFY_AMQP_ENABLE_<service>`, because if the credentials are wrong two things happen:
- Minio will NOT write any authentication errors or failures in the logs
- Minio's first process will not close properly generating the error mentioned in this issue about ports still occupied by Minio's previous process, it seems that authentication failure leads to zombie status for the first process or taking longer to close and release the ports

**Comment by ngochai711** [comment]
> Changes are already available in the latest version: `bitnami/minio:2021.10.13-debian-10-r0`.
> 
> Please let me know if it solves the issue.

Thank you. This solution for my minio
